machine/src/lib.rs

```diff
@@ -115,3 +158,10 @@ pub trait WithBalances: Machine {
 		_indirect: &[(Address, U256)],
 	) -> Result<(), Self::Error> { Ok(()) }
 }
+
+/// Metadata with total difficulty information.
+pub trait TotalDifficulty {
+	type Value: Ord + Add;
```

Comment:
this seems redundant with `TotalScore`

Reply:
Ooops looks like that trait is not used anywhere. Removing.